payment/tests/models/test_coupon_model.py

```diff
@@ -29,6 +29,53 @@ def test_coupon_create(coupon_effect):
     assert coupon.code is not None
     assert coupon.effects.first() == coupon_effect
 
+def test_custom_coupon_code_saved(coupon_effect):
+    custom_code = 'CUSTOMCODE'
+
+    coupon = Coupon.objects.create(
+        code=custom_code,
+        start_date=make_aware(datetime.datetime.now())
+    )
+    coupon.effects.set([coupon_effect])
+
+    assert coupon.code == custom_code
+
+def test_custom_coupon_code_not_saved_if_already_exists():
+    custom_code = 'CUSTOMCODE'
+    existing_coupon = Coupon.objects.create(
+        code=custom_code,
+        start_date=make_aware(datetime.datetime.now())
+    )
+
+    with pytest.raises(ValueError) as exception_info:
+        Coupon.objects.create(
+            code=custom_code,
+            start_date=make_aware(datetime.datetime.now())
+        )
+
+    assert str(exception_info.value) == f"Coupon with code {custom_code} already exists"
+
+def test_new_code_generated_if_code_already_exists():
```

Comment:
This test name doesn't seem to match its functionality? This test seems to be verifying the same thing as the previous test?